src/rt/test/func/memory/memory_rc.h

```diff
@@ -0,0 +1,102 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include "memory.h"
+
+namespace memory_rc
+{
+  constexpr auto region_type = RegionType::Rc;
+  using C = C1<region_type>;
+
+  void test_region_vector()
+  {
+    auto& alloc = ThreadAlloc::get();
+    auto* o = new (alloc) C;
+
+    const int NUM_OBJECTS = 128;
+
+    // Allocate a bunch of objects so that the region vector uses many
+    // allocation blocks to track each object in the region.
+    for (int i = 0; i < NUM_OBJECTS; i++)
+    {
+      auto* o1 = new (alloc, o) C;
+      assert(RegionRc::debug_get_ref_count(o1) == 1);
+    }
+
+    // Iterate through the region vector and make sure that all the
+    // refcounts make sense.
+    auto reg = (RegionRc*)Region::get(o);
+    RegionVector<ObjectCount, Alloc>::iterator iter(reg->get_trivial_vec());
+
+    ObjectCount* rm1 = nullptr;
+    ObjectCount* rm2 = nullptr;
+    ObjectCount* rm3 = nullptr;
+
+    int count = 0;
+    for (auto object_count : iter)
+    {
+      assert(object_count->count == 1);
+
+      if (count == 5)
+        rm1 = object_count;
+
+      if (count == 40)
+        rm2 = object_count;
+
+      if (count == 100)
+        rm3 = object_count;
+      count++;
+    }
+
+    assert(count == NUM_OBJECTS);
+
+    reg->get_trivial_vec()->remove(rm3);
```

Comment:
Could we expose a more useful primitive on the region.
```
RegionRC::decref(RegionRC*, Object*)
RegionRC::incref(RegionRC*, Object*)
```
The current API requires too much knowledge of the implementation

Reply:
This also makes me less certain of separating the two stacks.  If we have to branch on which stack for each incref and decref, then this is going to potentially cost way more than we save at teardown.

Reply:
Your right.  I guess it is only for decref, where you hit empty, and then you have to put it on the relevant free list, but then you have to inspect the header anyway.